Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStorePrivate.h

```diff
@@ -155,7 +155,8 @@ typedef NS_ENUM(uint8_t, _WKRestrictedOpenerType) {
 
 - (void)_runningOrTerminatingServiceWorkerCountForTesting:(void(^)(NSUInteger))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
 
-- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_SWIFT_UI_ACTOR void (^)(NSData *))completionHandler;
+- (void)_fetchDataOfTypes:(NSSet<NSString *> *)dataTypes completionHandler:(WK_SWIFT_UI_ACTOR void(^)(NSData *))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
+- (void)_restoreDataOfTypes:(NSSet<NSString *> *)dataTypes data:(NSData *)data completionHandler:(WK_SWIFT_UI_ACTOR void(^)(void))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
```

Comment:
I think the types that are fetched are encoded in the `_fetchDataOfTypes` step above, and the restore step should just ingest them all.
To many weird edge cases arise from allowing the client to choose a different set of types on the restore step, and I'm not sure it's useful to do.

Reply:
That's a good point. I've updated the fetch step to encode the data type and the restore step to just ingest them.